src/ccl_redshifts.c

```diff
@@ -171,109 +170,90 @@ struct norm_params {
 };
 
 static double ccl_norm_integrand(double z, void* params)
-{	
-	
+{
+
   // This is a struct that contains a true redshift and a pointer to the information about the photo_z model
   struct pz_params pz_val_p; // parameters for the photoz pdf wrt true-z
-  
+
   double pz_int=0; // pointer to the value of the integral over the photoz model
   struct norm_params *p = (struct norm_params *) params; // parameters of the current function (because of form required for gsl integration)
-  
+
   double z_min = p->bin_zmin_;
   double z_max = p->bin_zmax_;
-  
-  // Check whether ccl_splines and ccl_gsl exist; exit gracefully if they 
-  // can't be loaded
-  if(ccl_splines==NULL || ccl_gsl==NULL) ccl_cosmology_read_config();
-  if(ccl_splines==NULL || ccl_gsl==NULL) {
-    ccl_raise_exception(CCL_ERROR_MISSING_CONFIG_FILE, 
-                        "ccl_redshift.c: Failed to read config file.");
-    return NAN;
-  }
-  
+
   // Set up parameters for the pz part of the intermediary integral.
   pz_val_p.z_true = z;
   pz_val_p.status = p->status;
   pz_val_p.pz_information = p-> pz_information;
-  
-  // Do the intermediary integral over the model relating  photo-z to true-z	
-  gsl_integration_cquad_workspace * workspace = gsl_integration_cquad_workspace_alloc(ccl_gsl->N_ITERATION);
+
+  // Do the intermediary integral over the model relating  photo-z to true-z
+  gsl_integration_cquad_workspace * workspace = gsl_integration_cquad_workspace_alloc(RDSH_GSL_N_ITERATION);
```

Comment:
Same for these: why are these not part of the new structs?

Reply:
OK, I think I get it: none of these functions take a cosmology, and since we're thinking about scrapping them, it's not worth complicating our lives further. Correct?

Reply:
Yes this is correct!